src/main/java/ch/njol/skript/util/slot/ThrowableProjectileSlot.java

```diff
@@ -0,0 +1,68 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.util.slot;
+
+import org.bukkit.Material;
+import org.bukkit.entity.ThrowableProjectile;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.registrations.Classes;
+
+/**
+ * Represents the displayed item of a throwable projectile.
+ */
+public class ThrowableProjectileSlot extends Slot {
+	
+	private ThrowableProjectile projectile;
+	
+	public ThrowableProjectileSlot(ThrowableProjectile projectile) {
+		this.projectile = projectile;
+	}
+	
+	@Override
+	public ItemStack getItem() {
+		return projectile.getItem();
+	}
+	
+	@Override
+	public void setItem(@Nullable ItemStack item) {
+		projectile.setItem(item != null ? item : new ItemStack(Material.AIR));
+	}
+	
+	@Override
```

Comment:
correct me if i'm wrong, but couldn't you do `projectile.getItem().getAmount()` and `project.getItem().setAmount(amount)`

Reply:
I could, but the amount is always 1, so I decided to be consistent with the [ItemFrameSlot](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/util/slot/ItemFrameSlot.java#L51) class.

Reply:
Looks like @ShaneBeee [wrote that line](https://github.com/SkriptLang/Skript/blame/d246107d182ef1566016238c6c8d4bd199c6af00/src/main/java/ch/njol/skript/util/slot/ItemFrameSlot.java#L56) and @FranKusmiruk actually brought up the same concern that I did but was resolved. In my opinion, Skript should allow the same access to implemented APIs as Java would (with the exception of stopping you from doing things that break the server). what do you guys think @Mr-Darth @FranKusmiruk @ShaneBeee?

Reply:
While I normally agree that Skript should be more consistent with Bukkit, Skript is often doing things quite different than Bukkit would.
Also, if Im not mistaken, this method is generally used internally and not by the end user. I could be wrong.
Regardless, attempting to set the item stack amount to anything about 1, will just get ignored either way.

Reply:
As Shane said, the item amount can't be any number but 1, so everything will work the same, no matter how it's implemented.